internal/runners/clean/run.go

```diff
@@ -3,33 +3,63 @@ package clean
 import (
 	"os"
 
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
 	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/runners/prepare"
 )
 
 func (u *Uninstall) runUninstall() error {
 	err := removeCache(u.cfg.CachePath())
 	if err != nil {
-		return err
+		u.reportError(locale.Tl("uninstall_remove_cache_err", "Failed to remove cache directory."), err)
 	}
 
-	err = removeInstall(u.installPath)
+	err = removeInstall(u.cfg, u.installDir)
 	if err != nil {
-		return err
+		u.reportError(locale.Tl("uninstall_remove_executables_err", "Failed to remove all State Tool files in installation directory {{.V0}}", u.installDir), err)
 	}
 
 	err = removeConfig(u.cfg)
 	if err != nil {
-		return err
+		u.reportError(locale.Tl("uninstall_remove_config_err", "Failed to remove configuration directory {{.V0}}", u.cfg.ConfigPath()), err)
+	}
+
+	err = undoPrepare()
+	if err != nil {
+		u.reportError(locale.Tl("uninstall_prepare_err", "Failed to undo some installation steps."), err)
 	}
 
 	u.out.Print(locale.T("clean_success_message"))
 	return nil
 }
 
+func (u *Uninstall) reportError(msg string, err error) {
+	logging.Error("%s: %v", msg, errs.Join(err, ": "))
+	u.out.Notice(msg)
+}
```

Comment:
If we encountered errors during the uninstall we may want some sort of messaging at the end telling the user what went wrong and how to fix it. Either removing it manually or fixing something and attempting the command again. I guess this could also be part of the individual error notices.